h2o-r/h2o-package/R/mli.R

```diff
@@ -0,0 +1,2209 @@
+                                      color = .data$name,
+                                      text = .data$text),
+                         data = results) +
+      stat_count_or_bin(!is.numeric(newdata[[column]]),
+        ggplot2::aes(x = .data[[col_name]], y = (.data$..count.. / max(.data$..count..)) * diff(y_range) / 1.61),
+        position = ggplot2::position_nudge(y = y_range[[1]] - 0.05 * diff(y_range)), alpha = 0.2,
+        inherit.aes = FALSE, data = as.data.frame(newdata[[column]])) +
+      geom_point_or_line(!is.numeric(newdata[[column]]), ggplot2::aes(group = .data$name)) +
+      geom_point_or_line(!is.numeric(newdata[[column]]),
+                         if (is.factor(pdp[[col_name]])) {
+                           ggplot2::aes(shape = "Partial Dependence", group = "Partial Dependence")
+                         } else {
+                           ggplot2::aes(linetype = "Partial Dependence", group = "Partial Dependence")
+                         },
+                         data = pdp, color = "black"
+      ) +
+      ggplot2::geom_rug(ggplot2::aes(x = .data[[col_name]], y = NULL, text = NULL),
+                        sides = "b", alpha = 0.1, color = "black",
+                        data = stats::setNames(as.data.frame(newdata[[column]]), col_name)
+      ) +
+      ggplot2::labs(y = "Response", title = sprintf(
+        "Individual Conditional Expectations on \"%s\"%s\nfor Model: \"%s\"", col_name,
+        if (is.null(target)) {
+          ""
+        } else {
+          sprintf(" with Target = \"%s\"", target)
+        },
+        model@model_id
+      )) +
+      ggplot2::scale_color_viridis_d(option = "plasma") +
+      ggplot2::scale_linetype_manual(values = c("Partial Dependence" = "dashed")) +
+      # make the histogram closer to the axis. (0.05 is the default value)
+      ggplot2::scale_y_continuous(expand = ggplot2::expansion(mult = c(0, 0.05))) +
+      ggplot2::theme_bw() +
+      ggplot2::theme(
+        legend.title = ggplot2::element_blank(),
+        axis.text.x = ggplot2::element_text(angle = if (h2o.isfactor(newdata[[col_name]])) 45 else 0, hjust = 1),
+        plot.margin = margin
+      )
+    return(p)
+  })
+}
+
+######################################## Explain ###################################################
+
+#' Generate explanations for \code{object} on \code{test_frame}.
+#'
+#' @param object One of the following: an H2OAutoML, an H2OAutoML Leaderboard slice, a model, a list of models.
+#' @param test_frame An H2OFrame.
+#' @param columns_of_interrest A vector of feature names. If specified, plot just these columns.
+#' @param include_explanations If specified, do only the specified explanations.
+#'   (Mutually exclusive with exclude_explanations)
+#' @param exclude_explanations Exclude specified explanations.
+#' @param top_n_features If \code{columns_of_interrest} is missing, use top n features.
+#' @param best_of_family If True, explain only best of family models
```

Comment:
@tomasfryda Can you explain this parameter a bit more -- can you fill in the missing pieces:  "If True, explain only the best model from each algorithm class; if False ... "

Reply:
That parameter now applies only to the PDP plot - by default it's showing just the best of family. 
I wonder if I should remove the parameter from the `h2o.explain` - it would still be overridable by the `plot_overrides`.
Or I could use just best of family for the rest of the "multi-model" explanations - model correlation heat map and variable importance heat map - now it uses top 20 models which seems to me to be better so I would probably go for the first option.

What do you think @ledell ?

Reply:
> I wonder if I should remove the parameter from the `h2o.explain` - it would still be overridable by the `plot_overrides`.

Just noticed that in Python version I don't expose this parameter in `h2o.explain` (but the user can still choose to show all models by using the `plot_overrides`)

Reply:
I removed this parameter and made it smart as is in python: 
When explaining AutoML, show just best of family to get some general overview, but when explaining a list of models assume that the user actually handpicked the models and actually wants to see them all.

This is now a default behaviour but it is still overridable by using the `plot_overrides`.